app/models/user.rb

```diff
@@ -78,6 +78,8 @@ class User < ApplicationRecord
 
   validates :username, presence: true, if: :username_required?
   validates :username, uniqueness: { scope: :registering_with_oauth }, if: :username_required?
+  validates :email, presence: true, 'valid_email_2/email': true, allow_blank: true, unless: :email_consul_dev?
```

Comment:
Hi, Pierre!

This issue is very tricky. I thought about adding the `presence` rule a few months ago, but then realized there were dozens (maybe hundreds? I don't remember :thinking:) of users in production databases with no emails. I don't remember the circumstances allowing blank emails: maybe it was related to users belonging to organizations, or maybe it was related to managed users.

Since we need to study this issue carefully and make sure there are no side effects in existing installations, I've added the `post-1.1` label, so this pull request will not be included in release 1.1.

Reply:
There are definitely blank e-mails for managed users! I think `allow_blank: true` should work for these? But I guess `presence: true` shouldn't be here.